scripts/service-update-scripts/dcos-commons-merge.sh

```diff
@@ -0,0 +1,200 @@
+input_template_image_dir=$REPO_NAME/docs/pages/img/services
+output_template_image_dir=$output_template_dir/img
+rm -rf $output_template_image_dir
+mkdir -p $output_template_image_dir
+cp -v $input_template_image_dir/* $output_template_image_dir
+ls $input_template_image_dir/* >> $merge_list_file
+
+# Copying finished, delete source repo:
+rm -rf $REPO_NAME/
+
+###
+# Conversion hacks
+
+mangle_content_file() {
+    echo "Mangling $1..."
+
+    original=$1
+    target=$original.new
+    cp $original $target
+
+    # Insert source comment, and these headers: "model: /services/$package/data.yml" + "render: mustache"
+    awk -v n=2 "/---/ { if (++count == n) sub(/---/, \"model: /services/$package/data.yml\nrender: mustache\n---\n\n<!-- Imported from $REPO_URL:$branch -->\"); } 1{print}" $target > tmp && mv tmp $target
+
+    # Link/image url hacks:
+
+    # Remove https://docs.mesosphere.com from links (use , as sed delim)
+    sed -i 's,https://docs.mesosphere.com/,/,g' $target
+
+    # Update template image urls: /dcos-commons/img/services/ => services/include/img/ (use , as sed delim)
+    sed -i 's,/dcos-commons/img/services/,/services/include/img/,g' $target
+
+    # Update service image urls: /dcos-commons/services/$framework/img/ => /services/$package/$version/img/ (use , as sed delim)
+    sed -i "s,/dcos-commons/services/$framework/img/,/services/$package/$version/img/,g" $target
+
+    # Jekyll templating hacks for service .md files:
+
+    # Remove {% assign data = site.data.services.svcname %}
+    sed -i 's/{% assign .* %}$//g' $target
+
+    # Replace {% include services/foo.md data=data %} => #include /services/include/foo.tmpl (use , as sed delim)
+    sed -i 's,{% include services/\(.*\)\.md .*%},#include /services/include/\1.tmpl,g' $target
+
+    # Replace {{ data.foo }} => {{ model.foo }} (THIS MUST HAPPEN AFTER include.data.foo ABOVE)
+    sed -i 's/{{ \?data\.\([^ ]*\) \?}}/{{ model.\1 }}/g' $target
+
+    # Jekyll templating hacks for common .tmpl files:
+
+    # Replace {{ include.data.foo }} => {{ model.foo }}
+    sed -i 's/{{ \?include\.data\.\([^ ]*\) \?}}/{{ model.\1 }}/g' $target
+
+    #diff $original $target || true
+    mv $target $original
+}
+
+for filepath in $(find $output_package_version_dir -iname "*.md"); do
+    mangle_content_file $filepath
+done
+
+for filepath in $(find $output_template_dir -iname "*.tmpl"); do
+    mangle_content_file $filepath
+done
+
+# Update packageName in data.yml (including or omitting beta- prefix)
+sed -i "s/packageName:.*/packageName: $package/g" $output_yml_data_filepath
+
+# Add title/version information to latest index file
+sed -i "s/navigationTitle:.*/navigationTitle: $title $version/g" ${output_package_version_dir}/index.md
+sed -i "s/title:.*/title: $title $version/g" ${output_package_version_dir}/index.md
+
+echo "------------------------------"
+echo " $title $version merge complete"
+echo "------------------------------"
```

Comment:
@nickbp can you please run `shellcheck` against this script and make corrections on some "red" comments. I've commented some of them under the exact lines. There are very few major things to correct, the most of suggested corrections are minor things, but this will ensure us that the build job will not make weird issues, infuture.

Reply:
Done. Ended up just cleaning up all the errors.